utils/flare.py

```diff
@@ -206,15 +211,55 @@ def _add_conf_tar(self):
             if self._can_read(file_path, output=False):
                 self._add_clean_confd(file_path)
 
+    # Collect JMXFetch-specific info and save to jmxinfo directory if jmx config
+    # files are present and valid
+    def _add_jmxinfo_tar(self):
+        _, _, should_run_jmx = self._capture_output(self._should_run_jmx)
+        if should_run_jmx:
+            # status files (before listing beans because executing jmxfetch overwrites status files)
+            for file_name, file_path in [
+                (JMXFiles._STATUS_FILE, JMXFiles.get_status_file_path()),
+                (JMXFiles._PYTHON_STATUS_FILE, JMXFiles.get_python_status_file_path())
+            ]:
+                if self._can_read(file_path, warn=False):
+                    self._tar.add(
+                        file_path,
+                        os.path.join(self._prefix, 'jmxinfo', file_name)
+                    )
+
+            # beans lists
+            for command in ['list_matching_attributes', 'list_everything']:
+                log.info("  * datadog-agent jmx {0} output".format(command))
+                self._add_command_output_tar(
+                    os.path.join('jmxinfo', '{0}.log'.format(command)),
+                    partial(self._jmx_command_call, command)
+                )
+
+            # java version
+            log.info("  * java -version output")
+            self._add_command_output_tar(
+                os.path.join('jmxinfo', 'java_version.log'),
+                # We use lambda so that JMXFetch.get_configuration is evaluated in _add_command_output_tar,
+                # which captures the logging output from JMXFetch
+                lambda: self._java_version(JMXFetch.get_configuration(get_confd_path())[2])
```

Comment:
Nitpicky: we are also adding the output of `JMXFetch.get_configuration(get_confd_path())` to `java_version.log`. Is it intended ?

```
>>>> STDOUT <<<<
tomcat check does not have a valid JMX configuration: You need to have at least one instance defined in the YAML file for this check
Popen(['java', '-version'], stderr = -2, stdout = -1) called
java version "1.7.0_51"
Java(TM) SE Runtime Environment (build 1.7.0_51-b13)
Java HotSpot(TM) 64-Bit Server VM (build 24.51-b03, mixed mode)

>>>> STDERR <<<<
```
Maybe we could drop it and add a `>>>> CMD <<<<` section to `_add_command_output_tar ` with the actual command printed (in this case it's interesting as it may contains a custom java path).